Lib/importlib/util.py

```diff
@@ -84,11 +84,16 @@ def find_spec(name, package=None):
     if fullname not in sys.modules:
         parent_name = fullname.rpartition('.')[0]
         if parent_name:
-            # Use builtins.__import__() in case someone replaced it.
-            parent = __import__(parent_name, fromlist=['__path__'])
-            return _find_spec(fullname, parent.__path__)
+            try:
+                parent_path = __import__(parent_name,
+                                         fromlist=['__path__']).__path__
+            except AttributeError as e:
+                raise ModuleNotFoundError(
```

Comment:
Is the `AttributeError` purely from the attempt to access `__path__` on the imported module? If so then the `__import__()` call should go outside the `try` block to limit the scope of what will have an `AttributeError` caught. If there's another potential trigger of the `AttributeError` then we have a bigger problem. ;)